### PR TITLE
This passes the URL along with the target for reference.

### PR DESCRIPTION
**Pull request by Camwyn**
Goes along with https://github.com/mt-support/tribe-ext-external-links-in-new-tab/pull/4

While tweaking the extension, I realized we aren't passing the URL - so there is no way to know what/why you'd change the target.

[EXT-151]

[EXT-151]: https://moderntribe.atlassian.net/browse/EXT-151

### CODE REVIEW

File: src/functions/template-tags/organizer.php [hunk 1]
```diff
@@ -267,16 +267,18 @@ function tribe_get_organizer_link( $post_id = null, $full_link = true, $echo = f
 
 		$org_id = tribe_get_organizer_id( $post_id );
 		if ( class_exists( 'Tribe__Events__Pro__Main' ) && get_post_status( $org_id ) == 'publish' ) {
+
+			$url    = esc_url_raw( get_permalink( $org_id ) );
```
Comment (tr1b0t) [inline src/functions/template-tags/organizer.php:271]: :warning: Equals sign not aligned correctly; expected 1 space but found 4 spaces

<sub>hash: d434d99842d9e86e6e22adac05ebd110</sub>

Comment (Camwyn) [reply src/functions/template-tags/organizer.php:271]:
```suggestion
			$url = esc_url_raw( get_permalink( $org_id ) );
```

**Comment by paulmskim** [comment]
I wonder if it makes sense to pass the event id (or organizer or venue id) or the event (organizer, venue) object instead, the case I can think of is if someone wants to change the target based on another property, not just the url.

**Comment by Camwyn** [comment]
> I wonder if it makes sense to pass the event id (or organizer or venue id) or the event (organizer, venue) object instead, the case I can think of is if someone wants to change the target based on another property, not just the url.

I wouldn't hurt. We could pass both - post ID as a third param?
@bordoni @lucatume Thoughts?

**Comment by bordoni** [comment]
I like that for sure. Lets get that going as well. Please add changelogs for all of those changes.

File: src/functions/template-tags/organizer.php [hunk 2]
```diff
@@ -355,8 +358,8 @@ function tribe_get_organizer_website_url( $postId = null ) {
 	 *
 	 * Returns the event Organizer Name with a link to their supplied website
 	 *
-	 * @param $post_id post ID for an event
-	 * @param $label   text for the link
+	 * @param $post_id post ID for an event.
```
Comment (tr1b0t) [inline src/functions/template-tags/organizer.php:361]: :no_entry_sign: Missing parameter name

<sub>hash: 31071a26b80ea736c57e2d373b244902</sub>

File: src/functions/template-tags/organizer.php [hunk 3]
```diff
@@ -355,8 +358,8 @@ function tribe_get_organizer_website_url( $postId = null ) {
...
+	 * @param $label   text for the link.
```
Comment (tr1b0t) [inline src/functions/template-tags/organizer.php:362]: :no_entry_sign: Missing parameter name

<sub>hash: 9d2d54521121f38a77fde257ec1153f3</sub>

File: src/functions/template-tags/venue.php [hunk 4]
```diff
@@ -710,12 +710,13 @@ function tribe_get_venues( $only_with_upcoming = false, $posts_per_page = -1, $s
 	 *
 	 * @since ??
 	 *
-	 * @param null $post_id
-	 * @param null $label
+	 * @param null|int    $post_id
```
Comment (tr1b0t) [inline src/functions/template-tags/venue.php:713]: :no_entry_sign: Missing parameter comment

<sub>hash: d626d0b7f09c63d34dac33d1cafe148f</sub>

File: src/functions/template-tags/venue.php [hunk 5]
```diff
@@ -710,12 +710,13 @@ function tribe_get_venues( $only_with_upcoming = false, $posts_per_page = -1, $s
...
+	 * @param null|string $label
```
Comment (tr1b0t) [inline src/functions/template-tags/venue.php:714]: :no_entry_sign: Missing parameter comment

<sub>hash: 1c907fcbeaa2af6d8024a1052bf58e4a</sub>

File: src/functions/template-tags/venue.php [hunk 6]
```diff
@@ -778,7 +780,7 @@ function tribe_get_venue_website_link( $post_id = null, $label = null ) {
 	 *
 	 * @since ??
 	 *
-	 * @param int|null $post_id
+	 * @param int|null $post_id The event ID
```
Comment (tr1b0t) [inline src/functions/template-tags/venue.php:783]: :no_entry_sign: Parameter comment must end with a full stop

<sub>hash: c15e40c1a51c88be898c7e7ba7a65e1f</sub>

File: src/views/blocks/event-website.php [hunk 7]
```diff
@@ -22,13 +23,15 @@
  *
  * @since 5.1.0
  *
- * @param string the target attribute string. Defaults to "_self".
+ * @param string $target The target attribute string. Defaults to "_self".
+ * @param string $href   The link URL.
  */
-$target = apply_filters( 'tribe_get_event_website_link_target', '_self' );
+$target = apply_filters( 'tribe_get_event_website_link_target', '_self', $href, $this->context[ 'post_id' ] );
```
Comment (tr1b0t) [inline src/views/blocks/event-website.php:29]: :no_entry_sign: Array keys must NOT be surrounded by spaces if they only contain a string or an integer.

<sub>hash: f65c9ae2339dbf681684837cb09a791e</sub>

File: src/functions/template-tags/organizer.php [hunk 8]
```diff
@@ -221,9 +221,9 @@ function tribe_has_organizer( $postId = null ) {
 	/**
 	 * Organizer Email
 	 *
-	 * Returns the Organizer's Email
+	 * Returns the Organizer's Email.
 	 *
-	 * @param int $postId Can supply either event id or organizer id, if none specified, current post is used
+	 * @param int $postId       Either event id or organizer id, if none specified, current post is used.
```
Comment (tr1b0t) [inline src/functions/template-tags/organizer.php:226]: :no_entry_sign: Expected 2 spaces after parameter type; 1 found

<sub>hash: ab180c522898f60473fb589cc2faba5a</sub>

File: src/functions/template-tags/venue.php [hunk 9]
```diff
@@ -831,9 +834,9 @@ function tribe_get_venue_details( $post_id = null ) {
 	 *
 	 * @since ??
 	 *
-	 * @param int $event_id Event ID
-	 * @param boolean $link Whether or not to wrap the text in a venue link
-	 * @return string
+	 * @param int $event_id The event ID.
```
Comment (tr1b0t) [inline src/functions/template-tags/venue.php:837]: :no_entry_sign: Expected 5 spaces after parameter type; 1 found

<sub>hash: ff6dbc539d9e88de64ecb1b00227a6c0</sub>

File: src/functions/template-tags/venue.php [hunk 10]
```diff
@@ -571,8 +572,8 @@ function tribe_get_coordinates( $postId = null ) {
 	 *
 	 * @since ??
 	 *
-	 * @param int $postId Can supply either event id or venue id, if none specified, current post is used
-	 * @return bool Depending on the venue checkbox of overwrite coordinates
```
Comment (Camwyn) [inline src/functions/template-tags/venue.php:575]: This one doesn't really make sense either?

File: src/functions/template-tags/venue.php [hunk 11]
```diff
@@ -614,7 +615,7 @@ function tribe_get_phone( $postId = null ) {
 		 * @since ??
 		 * @since 4.5.11 Added docblock and venue ID to filter
 		 *
-		 * @param bool $output Whether the venue's coordinates are overwritten or not.
```
Comment (Camwyn) [inline src/functions/template-tags/venue.php:617]: This was just wrong.
This one is what prompted going through them all.

**Comment by Camwyn** [comment]
@bordoni I had to fix conflicts - needs another approval 🙏

**Pull request merged**